ironfish/src/wallet/wallet.ts

```diff
@@ -991,6 +992,16 @@ export class Wallet {
       )
     }
 
+    for (const output of options.outputs ?? []) {
+      if (output.memo.byteLength > MEMO_LENGTH) {
+        throw new Error(
+          `Memo is too long, max byte length is ${MEMO_LENGTH}, provided memo is ${
+            output.memo.byteLength
+          } bytes. Memo: ${output.memo.toString('utf8')}`,
```

Comment:
Since we're taking memos as buffers in places now, I wonder if we should display it as something other than utf8, like hex... Or what do you think about instead saying something like `The memo fields on the following outputs are more than 32 bytes: 1, 2, ....`

Also we should probably throw this as a breaking change in the release notes draft

Reply:
You could display the both the hex and the decoded string (if it is a valid utf-8 string). Examples

valid utf-8 string:

```
Memo: 74686973206d656d6f206973206c6f6e676572207468616e2033322063686172616374657273 ("this memo is longer than 32 characters")
```

invalid utf-8 string:

```
Memo: aa006cd60a64ae8b3ab4c80c579f47ed6648a2dcfa7b2fcee56d0ac7c9b229006b4b27a9daa5
```

We could also truncate long strings.

Regardless, if we plan to display user-inputted strings, I suggest we encode it to JSON first, so that special characters (like new line characters or ANSI escape sequences) do not result in weird output.

Reply:
We are currently truncating, but I don't really like it because it is easy for someone to not catch it is happening if they aren't paying attention to logs/ouputs.

I will with Derek's option, because if there are multiple error outputs then the message could get long and noisy. I think I would rather know which are the problematic outputs rather than have to recreate transaction having modified one of the outputs just to find that another of my outputs was a problem.